service/controller/rolebindingtemplate/resource/rolebinding/error.go

```diff
@@ -4,6 +4,15 @@ import (
 	"github.com/giantswarm/microerror"
 )
 
+var roleBindingCreationFailedError = &microerror.Error{
+	Kind: "roleBindingCreationFailedError",
+}
+
+// IsRoleBindingCreationFailed asserts roleBindingCreationFailedError.
+func IsRoleBindingCreationFailed(err error) bool {
+	return microerror.Cause(err) == roleBindingCreationFailedError
+}
+
```

Comment:
What will this be used for? The delete and status update cases simply mask the error returned by the k8s client, is there a reason for create/update to have a named error type? Either way, let's use the same approach everywhere